projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts

```diff
@@ -47,19 +47,27 @@ export class RasterLegendComponent implements OnInit, OnChanges {
     }
 
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.symbology) {
-            const symbology = changes.symbology.currentValue;
+        if (changes.layer) {
+            const symbology = changes.layer.currentValue.symbology;
             this.numberPipeParameters = RasterLegendComponent.calculateNumberPipeParameters(symbology.colorizer.breakpoints);
```

Comment:
I get an error message when using this branch:

> ERROR TypeError: breakpoints is undefined
    calculateNumberPipeParameters raster-legend.component.ts:58
    ngOnChanges raster-legend.component.ts:52
    Angular 36
    RxJS 5
    Angular 17
    RxJS 54
    changeProjectConfig project.service.ts:1006
    changeLayer project.service.ts:796
    RxJS 12
core.js:6479
    Angular 14
    RxJS 5
    Angular 17
    RxJS 54
    changeProjectConfig project.service.ts:1006
    changeLayer project.service.ts:796
    RxJS 23
    changeLayer project.service.ts:802
    toggleLegend project.service.ts:828
    toggleLegend layer-list.component.ts:132
    LayerListComponent_div_1_div_3_button_49_Template_button_click_0_listener layer-list.component.html:103
    Angular 23
    LayerListComponent_div_1_div_3_button_49_Template layer-list.component.html:103
    Angular 9
    LayerListComponent_div_1_div_3_Template layer-list.component.html:102
    Angular 6

Maybe this comes when you have a palette and call this method?

Reply:
It wasnt working for palette Colorizer bfd9bd28987dc5abefce0e66b801f16d51cefeca